app/controllers/api/current/blood_pressures_controller.rb

```diff
@@ -18,11 +18,29 @@ def merge_if_valid(blood_pressure_params)
       NewRelic::Agent.increment_metric('Merge/BloodPressure/schema_invalid')
       { errors_hash: validator.errors_hash }
     else
-      blood_pressure = BloodPressure.merge(Api::Current::Transformer.from_request(blood_pressure_params))
+      blood_pressure = ActiveRecord::Base.transaction do
+        set_patient_recorded_at(blood_pressure_params)
+        transformed_params = Api::Current::BloodPressureTransformer.from_request(blood_pressure_params)
+        BloodPressure.merge(transformed_params)
+      end
       { record: blood_pressure }
     end
   end
 
+  def set_patient_recorded_at(bp_params)
+    return if bp_params['recorded_at'].present?
+
+    patient = Patient.find_by(id: bp_params['patient_id'])
+    return if patient.blank?
+
+    patient.recorded_at = patient_recorded_at(bp_params, patient)
+    patient.save
+  end
+
+  def patient_recorded_at(bp_params, patient)
+    [bp_params['created_at'], patient.recorded_at].min
```

Comment:
`patient.recorded_at` can be `nil`? We don't constraint it anywhere yet.

Reply:
All patients will have `recorded_at` once the migrations are run. Also, current API will not be backdating `device_created_at`, we have to use `recorded_at`. But you are correct that this code will fail unless we run migrations.

We should use `min(bp_params['created_at'], patient.earliest_bp.recorded_at, patient.device_created_at)`

Reply:
actually this is confusing. we have changed it in cycles a couple of times. let's add comments this time.

Reply:
updated